keps/sig-network/2732-np-version/README.md

```diff
@@ -0,0 +1,803 @@
+        policy.
+
+      - If the policy uses featuers which are _known_ to the network
+        plugin but not fully _supported_ by it, the plugin will set
+        appropriate status conditions, as described below, indicating
+        that it may not have handled all of the policy's features in the
+        way the user intended.
+
+      - Otherwise the network plugin knows that it can correctly
+        implement the policy, and does so.
+
+#### The Supported Condition
+
+The `"Supported"` condition indicates whether the plugin supports the
+features used by a NetworkPolicy. Note that having a `{ type:
+"Supported", status: "False" }` condition does not necessarily mean that
+the policy is _completely_ unsupported.
+
+  - If the policy's `spec.minVersion` is a version that the plugin does
+    not recognize, then the plugin should set a status condition with `{
+    type: "Supported", status: "False", reason: "Version" }` and an
+    appropriate `message`
+
+  - If the policy has a recognized `minVersion`, but the plugin can see
+    that it uses features which the plugin does not support, then the
+    plugin should set a status condition with `{ type: "Supported",
+    status: "False", reason: "Unimplemented" }` and an appropriate
+    `message`.
+
+  - If a policy has a recognized `minVersion` and uses only supported
+    features, the plugin _may_ set a status condition with `{ type:
+    "Supported", status: "True" }`. However, this can also be assumed as
+    the default value if there is no `"Supported"` condition.
+
+Note the distinction between the first two cases; if the `minVersion` is
+higher than the highest version the plugin knows about, then the plugin
+has no way of telling whether it can implement the policy or not, and no
+way of knowing for sure if it's correctly interpreting even the parts
+that it can see. (For example, if a network plugin that only implemented
+version "1.9" received a policy with a combined `podSelector` and
+`namespaceSelector`, it might misinterpret it by reading the
+`namespaceSelector` but not the `podSelector`, and thereby allowing
+access to/from more pods than was intended.) Thus, it is best for it to
+not even try.
+
+On the other hand, if the plugin knows about a feature but doesn't
+implement it, they may still be able to implement the other parts of the
+policy safely. (For example, a plugin that does not implement egress
+policies at all might choose to still implement the ingress side of a
+mixed ingress/egress policy.)
+
+#### The Problem Condition
+
+A `"Problem"` condition with a status of `"True"` indicates some problem
+with the policy that the network plugin wants to report, which is not
+covered by any of the other conditions. eg:
+
+```
+conditions:
+  - type: Problem
+    status: True
+    reason: AmbiguousCIDR
+    message: "Interpreting 192.168.1.5/24 as 192.168.1.0/24 rather than 192.168.1.5/32"
+```
+
+#### Other Conditions
+
+```
+<<[UNRESOLVED matching-conditions ]>>
```

Comment:
ignoring the question of whether this is a good idea, it feels more like Status than Versioning